src/linter/block.go

```diff
@@ -210,7 +211,7 @@ func (b *BlockWalker) checkBinaryDupArgs(n, left, right node.Node) {
 	if !b.sideEffectFree(left) || !b.sideEffectFree(right) {
 		return
 	}
-	if astutil.NodeEqual(left, right) {
+	if b.NodeEqual(left, right) {
```

Comment:
This may now match `Foo == Bar` as duplicated if their values are identical.
It might look like a false positive for a user.
Perhaps we need to adjust the report message so it looks less broken for this kind of situation.